warrant/__init__.py

```diff
@@ -152,7 +152,7 @@ def __init__(
 
         self.user_pool_id = user_pool_id
         self.client_id = client_id
-        self.user_pool_region = self.user_pool_id.split('_')[0]
+        self.user_pool_region = user_pool_region if user_pool_region else self.user_pool_id.split('_')[0]
```

Comment:
I think we can get rid of the "user_pool_region" kwarg so this change is not necessary.

Reply:
But if it needs to change to maintain the API, then this logic is good.